### PR TITLE
Force serialization of primitive proto fields for block and tx data

### PR DESCRIPTION
**Pull request by ansnunez**
## Description

Proto3 does not serialize fields that are set to default value ([reference](https://developers.google.com/protocol-buffers/docs/proto3))

This is a breaking change from proto2, and will be problematic for any of our code that relies on byte consistency (e.g., hashes based on serialized blocks).

To fix this, we force the serialization of these fields by wrapping each one in a `oneof`. ([reference](https://stackoverflow.com/questions/37034746/do-not-set-default-values-in-optional-protobuf-fields-to-minimize-data-sent-over))

## Backward Compatibility

- [ ] This is not a breaking change
- [x] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libMessage/ZilliqaMessage.proto [hunk 1]
```diff
@@ -310,15 +310,15 @@ message ProtoAccountStore
 message ProtoPeer
 {
     ByteArray ipaddress    = 1;
-    uint32 listenporthost  = 2;
+    oneof oneof2 { uint32 listenporthost  = 2; }
 }
 
 message ProtoPoWSolution
 {
     uint64 nonce        = 1;
```
Comment (sandipbhoir) [inline src/libMessage/ZilliqaMessage.proto]: Any reason, we don't apply for `oneof` to `nonce` field?

Comment (ansnunez) [reply src/libMessage/ZilliqaMessage.proto]: > Any reason, we don't apply for `oneof` to `nonce` field?

I think a zero nonce for pow solution is highly unlikely?

Comment (sandipbhoir) [reply src/libMessage/ZilliqaMessage.proto]: In general, what is the reason for skipping other fields like `version` and `gaslimit`. These were optional in proto2 version previously.

Comment (KaustubhShamshery) [reply src/libMessage/ZilliqaMessage.proto]: so issue comes when the default value of these are actual values in the block (in this case the default value is 0). version starts from one and gaslimit is also non-zero.

**Pull request merged**